File: Easy/1022-Sum of Root To Leaf Binary Numbers/Iteration.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumRootToLeaf(self, root: TreeNode) -> int:
        nodes = [(root, 0)]
        nums = []

        while nodes:
            cur, num = nodes.pop()

            if not cur.left and not cur.right:
                nums.append((num << 1) + cur.val)
            if cur.left:
                nodes.append((cur.left, (num << 1) + cur.val))
            if cur.right:
                nodes.append((cur.right, (num << 1) + cur.val))

        return sum(nums)

File: Problemset/0258-Add Digits/Iteration.rs
impl Solution {
    pub fn add_digits(num: i32) -> i32 {
        let mut num = num;
        while num >= 10 {
            let mut tmp = 0;
            while num != 0 {
                tmp += num % 10;
                num /= 10;
            }
            num = tmp;
        }
        num
    }
}
